### DISCUSSION TITLE
Sleep tracker

### DISCUSSION THREAD

**Discussion by passthesh3ll** [iSoron/uhabits#980] [Feature Requests]
Hi, it could be very useful to add a specific tracker for sleep. We could add two input each day: _starting time_, _ending time_. So we can have a sort of sleep tracking like the bullet journal tracker:
![sss](https://user-images.githubusercontent.com/63223659/121856869-4749ba00-ccf5-11eb-95df-f93ca001340d.png)

**Comment by Silviopina** [comment]
Yes please!!! I am currently tracking went to bed time, got up time and total hours slept. @prestidigitonium solution would be great, having the app calculate & track time... it's SO CRUCIAL to a healthy life, perhaps #1!!!❤️

**Comment by hiqua** [comment]
Can you explain more what habit you're trying to form? I can think of these:
1. sleep duration ("I want to sleep at least 8h per night")
2. specific wake-up time / bed time if you want

but:

1. is covered by numerical habits
2. you can have a habit which you tick if you woke up within e.g. 15min of the time you wanted

**Comment by c-mena** [comment]
Related to #2296
